api/osrd_infra/views/rolling_stock.py

```diff
@@ -27,3 +35,83 @@ class LightRollingStockView(
     queryset = RollingStock.objects.order_by("id")
     serializer_class = LightRollingStockSerializer
     pagination_class = CustomPageNumberPagination
+
+
+class RollingStockLiveryView(
+    ViewSet,
+):
+    def retrieve(self, request, pk=None, rolling_stock_pk=None):
+        queryset = RollingStockLivery.objects.all()
+        livery = get_object_or_404(queryset, pk=pk)
+        image_db = livery.compound_image.image
+
+        if str(livery.rolling_stock.id) != rolling_stock_pk:
+            raise ValueError(f"Livery {pk} does not belong to rolling stock {rolling_stock_pk}")
+
+        stream = BytesIO(image_db)
+        image = Image.open(stream)
+
+        response = HttpResponse(content_type='image/png')
+        image.save(response, "PNG")
+        return response
+
+    def validate_create_request(self, rolling_stock_id, livery_name):
+        rolling_stock_queryset = RollingStock.objects.all()
+        get_object_or_404(rolling_stock_queryset, id=rolling_stock_id)
+        try:
+            otherLivery = RollingStockLivery.objects.get(rolling_stock_id=rolling_stock_id, name=livery_name)
+        except RollingStockLivery.DoesNotExist:
+            return
+        raise serializers.ValidationError(
+            f"an other livery (id {otherLivery.id}) with same name and rolling stock already exists"
+        )
+
+    def create_compound_image(self, images):
+        """
+        Create compound_image from separated images. Return an InMemoryUploadedFile.
+        """
```

Comment:
```suggestion
    def create_compound_image(self, images) -> InMemoryUploadedFile:
```
I think doc is paraphrasing the signature if the return type is added